operators/src/source/gdal_source.rs

```diff
@@ -820,771 +691,281 @@ mod tests {
+            params: GdalSourceParameters {
+                data_set: id.clone(),
             },
-            phantom_data: PhantomData,
-        };
-
-        let vres: Vec<Result<RasterTile2D<u8>, Error>> = gdal_source
-            .time_tile_iter(
-                origin_split_tileing_strategy,
-                TimeInterval::new_unchecked(0, 1000),
-                bounding_box,
+        }
+        .boxed();
+
+        let x_query_resolution = output_bounds.size_x() / output_shape.axis_size_x() as f64;
+        let y_query_resolution = output_bounds.size_y() / output_shape.axis_size_y() as f64;
+        let spatial_resolution =
+            SpatialResolution::new_unchecked(x_query_resolution, y_query_resolution);
+
+        let o = op.initialize(exe_ctx).unwrap();
+
+        o.query_processor()
+            .unwrap()
+            .get_u8()
+            .unwrap()
+            .query(
+                QueryRectangle {
+                    bbox: output_bounds,
+                    time_interval,
+                    spatial_resolution,
+                },
+                query_ctx,
             )
-            .map(|(time_interval, tile_information)| {
-                gdal_source.load_tile_data(time_interval, tile_information)
-            })
-            .collect();
-        assert_eq!(vres.len(), 6 * 4);
-        let upper_left_pixels: Vec<_> = vres
-            .into_iter()
-            .map(|t| {
-                let raster_tile = t.unwrap();
-                raster_tile
-                    .grid_array
-                    .get_at_grid_index([
-                        (tile_size_in_pixels[1] / 2) as isize,
-                        (tile_size_in_pixels[0] / 2) as isize,
-                    ])
-                    .unwrap() // pixel
-            })
-            .collect();
-
-        let ndvi_center_pixel_values = vec![
-            255, 255, 255, 255, 255, 255, 255, 116, 255, 54, 34, 255, 255, 255, 255, 212, 255, 145,
-            0, 0, 0, 0, 0, 0,
-        ];
-
-        assert_eq!(upper_left_pixels, ndvi_center_pixel_values);
+            .collect()
+            .await
     }
 
-    #[test]
-    fn test_iter_and_load_tile_data_bbox() {
-        let global_size_in_pixels = [1800, 3600];
-        let tile_size_in_pixels = [600, 600];
-        let dataset_upper_right_coord = (-180.0, 90.0).into();
-        let dataset_x_pixel_size = 0.1;
-        let dataset_y_pixel_size = -0.1;
-        let dataset_geo_transform = GeoTransform::new(
-            dataset_upper_right_coord,
-            dataset_x_pixel_size,
-            dataset_y_pixel_size,
-        );
-
-        let time_interval_provider = TimeIntervalInformation {
-            start_time: TimeInstance::from_millis(0),
-            time_step: TimeStep {
-                granularity: TimeGranularity::Seconds,
-                step: 1,
-            },
-        };
+    #[tokio::test]
+    async fn test_query_single_time_slice() {
+        let mut exe_ctx = MockExecutionContext::default();
```

Comment:
why dues this require a mut exe_ctx?

Reply:
because I add the data set info to it after creation, so the GdalSource can access it using the meta_data function